lib/utils.js

```diff
@@ -375,9 +375,29 @@ function parseSettings(urlStr) {
  *
  * @param {Object} target The target settings object
  * @param {Object} src The source settings object
+ * @param {Boolean} replaceTarget Should the src object replace the target value if it exists
  * @returns {Object} The merged settings object
  */
-function mergeSettings(target, src) {
+function mergeSettings(target, src, replaceTarget) { // NOTE: src is subclassSettings
+  const mergePolicy = {
+    description: {replace: true}, // string or array
+    options: {innerReplace: true}, // object
+    properties: {innerReplace: true}, // object
+    hidden: {replace: false}, // array
+    protected: {replace: false}, // array
+    indexes: {innerReplace: true}, // object
+    methods: {innerReplace: true}, // object
+    mixins: {innerReplace: true}, // object
+    // validations: {innerReplace: true}, // object // NOTE: not implemented
+    relations: {innerReplace: true},
+    scope: {replace: true}, // object
+    scopes: {innerReplace: true}, // object
+    acls: {replace: false}, // array
+    // any matching non array/object property is replaced
+    // any matching custom top-level object/array is replaced
+    default: {replace: true},
+  };
+
```

Comment:
the current code handling this is
```
if (srcValue == null || typeof srcValue !== 'object') {
  // The source item value is null, undefined or not an object
  dst[key] = srcValue;
}
```
**NOTE**: `src` reffers to subClass settings

it's pretty straight forward do we have extra configuration to add here?